File: apps/auth/src/services/value-hasher.ts
import { Injectable } from '@nestjs/common';
import * as bcrypt from 'bcryptjs';

import { IValueHasher } from '../contracts';

@Injectable()
export class ValueHasher implements IValueHasher {
  async hash(value: string): Promise<string> {
    const salt = await bcrypt.genSalt();
    return bcrypt.hash(value, salt);
  }

  async verify(value: string, valueHash: string): Promise<boolean> {
    return await bcrypt.compare(value, valueHash);
  }
}

File: src/Modules/Identity/Thrive.Modules.Identity.Infrastructure/Services/ValueHasher.cs
using Thrive.Modules.Identity.Application.Contracts;
using BC = BCrypt.Net.BCrypt;

namespace Thrive.Modules.Identity.Infrastructure.Services;

internal sealed class ValueHasher : IValueHasher
{
    public string Hash(string value)
    {
        return BC.HashPassword(value);
    }

    public bool Verify(string value, string hash)
    {
        return BC.Verify(value, hash);
    }
}